iree/compiler/Conversion/LinalgToLLVM/LinalgTileAndDistributePass.cpp

```diff
@@ -215,11 +113,39 @@ void LinalgTileAndDistributePass::runOnOperation() {
       }
     });
 
-    TileAndFuseOptions tileAndFuseOptions = {workgroupDistributionOptions,
-                                             allocateThreadLocalMemory};
-    if (failed(tileAndFuseLinalgBufferOps(funcOp, linalgOpsVec, dependenceGraph,
-                                          launchConfig, tileAndFuseOptions))) {
-      return signalPassFailure();
+    if (tiledLoops.empty()) {
+      linalg::LinalgLoopDistributionOptions workgroupDistributionOptions = {
+          [](OpBuilder &builder, Location loc,
+             ArrayRef<Range> parallelLoopRanges) {
+            auto numParallelDims = parallelLoopRanges.size();
+            SmallVector<linalg::ProcInfo, 3> procInfo(numParallelDims);
+            for (size_t dim = 0;
+                 dim < std::min(numParallelDims, static_cast<size_t>(3));
+                 ++dim) {
+              procInfo[numParallelDims - dim - 1] = {
+                  builder.createOrFold<IREE::HAL::InterfaceWorkgroupIDOp>(
+                      loc, builder.getIndexType(), APInt(64, dim)),
+                  builder.createOrFold<IREE::HAL::InterfaceWorkgroupCountOp>(
+                      loc, builder.getIndexType(), APInt(64, dim)),
+              };
+            }
+            return procInfo;
+          },
+          {linalg::DistributionMethod::CyclicNumProcsEqNumIters,
+           linalg::DistributionMethod::CyclicNumProcsEqNumIters,
+           linalg::DistributionMethod::CyclicNumProcsEqNumIters}};
```

Comment:
I feel this is too long to parse that it is the same statement. How about declare `procInfoCallBackFn` outside, so we make the init list shorter?

Reply:
This part of the code is probably ging to go away soon. Ill leave this as is for now.